diff -up ./rngd_entsource.c.altsource ./rngd_entsource.c
--- ./rngd_entsource.c.altsource	2017-05-31 09:27:12.531678280 -0400
+++ ./rngd_entsource.c	2017-05-31 09:29:28.252959165 -0400
@@ -179,6 +179,7 @@ static int discard_initial_data(struct r
 int init_entropy_source(struct rng *ent_src)
 {
 	struct sysfs_attribute *rngavail;
+	char buf[16];
 
 	ent_src->rng_fd = open(ent_src->rng_name, O_RDONLY);
 	if (ent_src->rng_fd == -1) {
@@ -186,6 +187,12 @@ int init_entropy_source(struct rng *ent_
 		return 1;
 	}
 
+	/* Try to read some data from the entropy source.  If it doesn't return
+ 	 * an error, assume its ok to use
+ 	 */
+	if (ent_src->xread(buf, sizeof(buf), ent_src) == 0)
+		goto source_valid;
+
 	/* RHEL7: since /dev/hwrng will exist now even if there isn't an rng backing it,
 	 * check to see if rng_available is empty, and return error if it is.
 	 */
@@ -208,6 +215,7 @@ int init_entropy_source(struct rng *ent_
 	}
 	sysfs_close_attribute(rngavail);
 
+source_valid:
 	src_list_add(ent_src);
 	/* Bootstrap FIPS tests */
 	ent_src->fipsctx = malloc(sizeof(fips_ctx_t));
