diff -up ./rngd.8.in.restore ./rngd.8.in
--- ./rngd.8.in.restore	2017-11-09 10:17:35.836384425 -0500
+++ ./rngd.8.in	2017-11-09 10:19:24.172965258 -0500
@@ -88,6 +88,20 @@ entropy pool.  Low values will hurt syst
 starves.  Do not set \fIfill-watermark\fR above the size of the
 entropy pool (usually 4096 bits).
 .TP
+\fB\-d\fI 1|0\fR, \fB\-\-no-drng=\fI1|0\fR
+Do not use drng as a source of random number input (default:0). Note, this
+option is deprecated, use -x instead
+.TP
+\fB\-n\fI 1|0\fR, \fB\-\-no-tpm=\fI1|0\fR
+Do not use tpm as a source of random number input (default:0). Note, this option
+is deprecated, use -x instead
+.TP
+\fB\-q\fR, \fB\-\-quiet\fR
+Suppress error messages
+.TP
+\fB\-v\fR, \fB\-\-verbose\fR
+Report available entropy sources, and exit
+TP
 \fB\-q\fR, \fB\-\-quiet\fR
 Suppress error messages
 .TP
diff -up ./rngd.c.restore ./rngd.c
--- ./rngd.c.restore	2017-11-09 10:17:35.837384422 -0500
+++ ./rngd.c	2017-11-09 10:20:39.048675553 -0500
@@ -79,8 +79,6 @@ static char doc[] =
 	"Check and feed random data from hardware device to kernel entropy pool.\n";
 
 static struct argp_option options[] = {
-	{ "debug", 'd', 0, 0, "Enable debug output" },
-
 	{ "foreground",	'f', 0, 0, "Do not fork and become a daemon" },
 
 	{ "ignorefail", 'i', 0, 0, "Ignore repeated fips failures" },
@@ -108,7 +106,13 @@ static struct argp_option options[] = {
 
 	{ "quiet", 'q', 0, 0, "Suppress error messages" },
 
-	{ "version" ,'v', 0, 0, "List rngd version" },
+	{ "verbose" ,'v', 0, 0, "Report available entropy sources" },
+
+	{ "no-drng", 'd', "1|0", 0,
+	  "Do not use drng as a source of random number input (default: 0)" },
+
+	{ "no-tpm", 'n', "1|0", 0,
+	  "Do not use tpm as a source of random number input (default: 0)" },
 
 	{ "entropy-count", 'e', "n", 0, "Number of entropy bits to support (default: 8), 1 <= n <= 8" },
 
@@ -182,9 +186,6 @@ static error_t parse_opt (int key, char
 {
 	long int idx;
 	switch(key) {
-	case 'd':
-		arguments->debug = true;
-		break;
 	case 'o':
 		arguments->random_name = arg;
 		break;
@@ -231,8 +232,24 @@ static error_t parse_opt (int key, char
 		arguments->quiet = true;
 		break;
 	case 'v':
-		printf("%s\n", argp_program_version);
-		return -EAGAIN;
+		arguments->list = true;
+		break;
+	case 'd': {
+		int n;
+		if ((sscanf(arg,"%i", &n) == 0) || ((n | 1)!=1))
+			argp_usage(state);
+		else
+			entropy_sources[ENT_RDRAND].disabled = true;
+		break;
+	}
+	case 'n': {
+		int n;
+		if ((sscanf(arg,"%i", &n) == 0) || ((n | 1)!=1))
+			argp_usage(state);
+		else
+			entropy_sources[ENT_TPM].disabled = true;
+		break;
+	}
 	case 'e': {
 		int e;
 		if ((sscanf(arg,"%i", &e) == 0) || (e < 0) || (e > 8))
diff -up ./rngd.h.restore ./rngd.h
