diff -up ./rngd.8.in.verbose ./rngd.8.in
--- ./rngd.8.in.verbose	2017-04-11 14:42:26.897634581 -0400
+++ ./rngd.8.in	2017-04-11 14:42:51.872477449 -0400
@@ -89,7 +89,7 @@ Do not use tpm as a source of random num
 Suppress error messages
 .TP
 \fB\-v\fR, \fB\-\-verbose\fR
-Report available entropy sources
+Report available entropy sources, and exit
 .TP
 \fB\-?\fR, \fB\-\-help\fR
 Give a short summary of all program options.
diff -up ./rngd.c.verbose ./rngd.c
--- ./rngd.c.verbose	2017-04-11 14:42:17.617692967 -0400
+++ ./rngd.c	2017-04-11 14:42:41.231544398 -0400
@@ -348,6 +348,7 @@ int main(int argc, char **argv)
 			printf("\tDRNG\n");
 		if (!rc_tpm)
 			printf("\tTPM\n");
+		return 1;
 	}
 
 	if (rc_rng
